src/DynamoCore/Engine/CodeGeneration/AstBuilder.cs

```diff
@@ -286,10 +281,10 @@ private void CompileToAstNodes(NodeModel node, List<AssociativeNode> resultList,
                     NodeModel inputModel = inputTuple.Item2;
                     AssociativeNode inputNode = inputModel.GetAstIdentifierForOutputIndex(outputIndexOfInput);
 
-#if DEBUG
-                    Validity.Assert(inputNode != null,
-                        "Shouldn't have null nodes in the AST list");
-#endif
+                    // If there are any null AST's (for e.g. if there's an error in the input node),
+                    // graph update for the given node is skipped.
+                    Validity.Assert(inputNode != null, "Shouldn't have null nodes in the AST list");
```

Comment:
will this print something to the console?

Reply:
@mjkkirschner this will not print to the console, which I think is okay as nodes getting into error state is a common scenario that may not need logging every time.

Reply:
I'm merging this in for now. Self-serve CI tests pass. Let me know if you have other comments later.